cmake/AddArterySubdirectory.cmake

```diff
@@ -0,0 +1,45 @@
+function(add_artery_subdirectory directory)
+    # include target subdirectory with certain artery
+    # extension (feature), if the following conditions match:
+    # 1) variable with id matching SWITCH is set to true (feature turned on)
+    # 2) packages that are listed in DEPENDENCIES are found
+    # 3) targets that are listed in REQUIRES are declared (used for extern modules)
```

Comment:
I think it is hard to distinguish "DEPENDENCIES" and "REQUIRES" because you could swap the mechanics and the terms still fit. I propose to use the target-based logic everywhere, which is the preferred approach in modern CMake.
Protobuf ships with the `protobuf::libprotobuf` target since CMake 3.9, the `pkg_check_modules` call creates a target if `IMPORTED_TARGET` is added (`PkgConfig::ZEROMQ`), and SEA_V2X provides the target `SEA_V2X::sea_v2x`.